rust-lib/src/lib.rs

```diff
@@ -194,14 +196,43 @@ enum RequestError {
     /// Outpoint parsing error: {_0}
     #[from]
     Outpoint(rgb::lnpbp::bitcoin::blockdata::transaction::ParseOutPointError),
+
+    /// I/O error: {_0}
+    #[from]
+    Io(std::io::Error),
+
+    /// Input error: {_0}
+    #[from]
+    Input(String),
+
+    /// Strict encoding error: {_0}
+    #[from]
+    StrictEncoding(rgb::lnpbp::strict_encoding::Error),
+}
+
+fn _get_consignment(
+    consignment_bytes: *const u8,
+    consignment_length: c_int,
+) -> Result<Consignment, RequestError> {
+    if consignment_bytes.is_null() {
+        return Err(RequestError::Input(s!(
+            "consignment bytes cannot be null"
+        )));
+    };
+    let consignment_bytes = unsafe {
+        slice::from_raw_parts(consignment_bytes, consignment_length as usize)
+    };
+    trace!("consignment bytes: {:x?}", consignment_bytes);
+    let consignment: Consignment = strict_decode(&consignment_bytes)?;
```

Comment:
this call triggers a `StrictEncoding(Io(UnexpectedEof))` error

Reply:
But this has nothing to do with the validation, as I read your initial report.

From which executable/demo this `_get_consignment` method is called so I can witness the failure?

Reply:
> But this has nothing to do with the validation, as I read from your initial report...

What initial report are you talking about? From what I see and remember, this was my initial report of the error: https://github.com/LNP-BP/rgb-sdk/pull/21#discussion_r527127722. Moreover since the `validate` method is failing (and was failing using the cli too) it's not wrong to say that validation is failing (implying that a step of it is failing, which seems to be the decoding part of it)

> From which executable/demo this _get_consignment method is called so I can witness the failure?

please look at the code introduced by this PR. the method `_get_consignment` (private method, so of course it's called from the same file) is called [here](https://github.com/LNP-BP/rgb-sdk/pull/21/files#diff-e5fe85b6ecefe2218ccf44990e99b82ccc92348897776f37eec3f97377306707R564) and [here](https://github.com/LNP-BP/rgb-sdk/pull/21/files#diff-e5fe85b6ecefe2218ccf44990e99b82ccc92348897776f37eec3f97377306707R525).

Use the node.js demo to test it